server/channels/app/ldap.go

```diff
@@ -71,6 +71,25 @@ func (a *App) TestLdapConnection(rctx request.CTX, settings *model.LdapSettings)
 	return nil
 }
 
+func (a *App) TestLdapFilters(rctx request.CTX, settings *model.LdapSettings) ([]model.LdapFilterTestResult, *model.AppError) {
+	license := a.Srv().License()
+	ldapI := a.LdapDiagnostic()
+
+	// NOTE: normally we would test (*a.Config().LdapSettings.Enable || *a.Config().LdapSettings.EnableSync),
+	// but we want to allow sysadmins to test the connection without enabling and saving the config first.
+	if ldapI != nil && license != nil && *license.Features.LDAP {
```

Comment:
This is already checked in the api layer. Just defence in depth or is there any other reasoning behind it?

Reply:
For this, we check at the api layer, check at the app layer. I'm guessing we're doing that in case its called from another place, we can't assume the check has already been done.

Reply:
I'm fine with it, except for one reason: it breaks the pattern -- every function in the api is following the same checks, so it feels a little strange to not for these two.  I'm 1/5 for keeping the checks because it doesn't cost anything (relatively -- this is as far from a hot path as you can get) and maintains consistency.

But I'll leave it to @esarafianou -- are there security implications for removing the permissions checks in the api layer, considering that all it does it call into the app layer, which reproduces the same checks?

Reply:
No concerns but I'm a 1/5 of keeping them for consistency reasons. Permission checks are in our API layer and, even if in this case we recheck in the app layer, being consistent helps with reviews and auditing.